### PR TITLE
Securing docker registry

### PR DESCRIPTION
**Pull request by pravisankar**

### CODE REVIEW

**Comment by pravisankar** [comment]
@adellape PTAL

File: admin_guide/docker_registry.adoc [hunk 1]
```diff
@@ -96,6 +97,113 @@ $ osc exec -p docker-registry-1-da73t find /registry
 <8> This directory contains links to all the layers this repository references
 <9> This file contains a reference to a specific layer that has been linked into this repository via an image
 
+== Securing Registry
```
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: Securing **the** Registry

File: admin_guide/docker_registry.adoc [hunk 2]
```diff
@@ -96,6 +97,113 @@ $ osc exec -p docker-registry-1-da73t find /registry
...
+$ osc get se docker-registry
+NAME              LABELS                                    SELECTOR                  IP(S)            PORT(S)
+docker-registry   docker-registry=default                   docker-registry=default   172.30.124.220   5000/TCP
+----
++
+. Create server certificates for the registry service IP and docker-registry.default.local
++
+[options="nowrap"]
+----
+$ osadm create-server-cert --signer-cert=ca.crt --signer-key=ca.key --signer-serial=ca.serial.txt --hostnames='docker-registry.default.local,172.30.124.220' --cert=registry.crt --key=registry.key
```
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: The hostname should be docker-registry.default.svc.cluster.local

File: admin_guide/docker_registry.adoc [hunk 3]
```diff
@@ -96,6 +97,113 @@ $ osc exec -p docker-registry-1-da73t find /registry
...
+  ...
+  - name: secret-volume
+    secret:
+      secretName: registry-secret
+...
+volumeMounts:
+  ...
+  - name: secret-volume
+    mountPath: /etc/secrets/
+    readOnly: true
```
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: @pmorie since this is a secret mount, is readOnly necessary?

File: admin_guide/docker_registry.adoc [hunk 4]
```diff
@@ -96,6 +97,113 @@ $ osc exec -p docker-registry-1-da73t find /registry
...
+----
++
+. Override registry tls config options by adding these environment variables to the registy deployment configuration.
```
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: Enable TLS by adding these environment variables to the deployment configuration:

File: admin_guide/docker_registry.adoc [hunk 5]
```diff
@@ -96,6 +97,113 @@ $ osc exec -p docker-registry-1-da73t find /registry
...
+  More details: https://github.com/docker/distribution/blob/master/docs/configuration.md#override-configuration-options[Overriding registry options]
++
+[options="nowrap"]
+----
+$ osc env dc/docker-registry REGISTRY_HTTP_TLS_CERTIFICATE=/etc/secrets/registry.crt REGISTRY_HTTP_TLS_KEY=/etc/secrets/registry.key
+----
++
+. Validate registry is running in tls mode.
```
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: TLS

File: admin_guide/docker_registry.adoc [hunk 6]
```diff
@@ -96,6 +97,113 @@ $ osc exec -p docker-registry-1-da73t find /registry
...
+[options="nowrap"]
+----
+$ osc get pods
+POD                       IP           CONTAINER(S)   IMAGE(S)                                  HOST                           LABELS                                                                                  STATUS    CREATED    MESSAGE
+docker-registry-1-da73t   172.17.0.1                                                            openshiftdev.local/127.0.0.1   deployment=docker-registry-4,deploymentconfig=docker-registry,docker-registry=default   Running   38 hours
+
+$ docker ps -a | grep docker-registry-1-da73t
+833a27ff9135        openshift/origin-docker-registry:v0.5.2   "/bin/sh -c 'REGISTR   37 hours ago        Up 37 hours                                   k8s_registry.33a55ebe_docker-registry-4-6vzeg_default_0ac743bf-042e-11e5-9401-080027c5bfa9_c088e612
+
+$ docker logs 833a27ff9135 2>&1 | grep tls
```
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: Use `osc log` or `osc logs` (can't remember which is current)

File: admin_guide/docker_registry.adoc [hunk 7]
```diff
@@ -96,6 +97,113 @@ $ osc exec -p docker-registry-1-da73t find /registry
...
+time="2015-05-27T05:05:53Z" level=info msg="listening on :5000, tls" instance.id=deeba528-c478-41f5-b751-dc48e4935fc2
+----
++
+. Copy CA certificate to the docker certs dir.
```
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: This must be done on all nodes.

File: admin_guide/docker_registry.adoc [hunk 8]
```diff
@@ -96,6 +97,113 @@ $ osc exec -p docker-registry-1-da73t find /registry
...
++
+----
+$ sudo mkdir -p /etc/docker/certs.d/172.30.124.220:5000
+$ sudo cp ca.crt /etc/docker/certs.d/172.30.124.220:5000
+
+$ sudo mkdir -p /etc/docker/certs.d/docker-registry.default.local:5000
```
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: Update name as per above comment

File: admin_guide/docker_registry.adoc [hunk 9]
```diff
@@ -96,6 +97,113 @@ $ osc exec -p docker-registry-1-da73t find /registry
...
+$ sudo cp ca.crt /etc/docker/certs.d/docker-registry.default.local:5000
+----
++
+. Remove `--insecure-registry` option from docker unit file (*/etc/sysconfig/docker*).
```
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: Only for this particular registry

File: admin_guide/docker_registry.adoc (hunk 3 above)
Comment (pmorie) [inline admin_guide/docker_registry.adoc]: You want readOnly, yes.

**Comment by pravisankar** [comment]
@adellape @ncdc updated, please review/merge

**Comment by adellape** [comment]
Thanks, @pravisankar. LGTM. If @ncdc doesn't have any further feedback, I'll merge.

**Comment by ncdc** [comment]
LGTM

**Pull request merged**